clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2829,6 +2829,56 @@ void CodeGenModule::CreateFunctionTypeMetadataForIcall(const FunctionDecl *FD,
       F->addTypeMetadata(0, llvm::ConstantAsMetadata::get(CrossDsoTypeId));
 }
 
+uint32_t
+CodeGenModule::calcRISCVZicfilpFuncSigLabel(const FunctionType &FT,
+                                            const bool IsCXXInstanceMethod,
+                                            const bool IsCXXVirtualMethod) {
```

Comment:
I assume this need use when we implement `%lpad_hash`, so this should implement under llvm folder rather than here.

Reply:
I moved just the hash part of this function to the llvm folder.
I suppose you're suggesting that the hash part needs to be moved to llvm to support implementing the `%lpad_label` asm operand modifier, which takes a mangled string and hash it into a 20-bit integer, so it's only the hashing part that needs to be re-located, not the whole mangling part.

Fixed.